network-api/networkapi/wagtailpages/pagemodels/buyersguide/article_page.py

```diff
@@ -1,5 +1,162 @@
 
+from networkapi.wagtailpages.pagemodels import customblocks
+from networkapi.wagtailpages.pagemodels.mixin import foundation_metadata
 
-class BuyersGuideArticlePage(wagtail_models.Page):
+
+class BuyersGuideArticlePage(
+    foundation_metadata.FoundationMetadataPageMixin,
+    wagtail_models.Page
+):
     parent_page_types = ['wagtailpages.BuyersGuideEditorialContentIndexPage']
+    subpage_types: list = []
+    template = 'pages/buyersguide/article_page.html'
+
+    hero_image = models.ForeignKey(
+        images.get_image_model_string(),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        help_text='Image for the hero section of the page.',
+    )
+    body = fields.StreamField(
+        block_types=(
+            ('accordion', customblocks.AccordionBlock()),
+            ('paragraph', blocks.RichTextBlock(
+                features=customblocks.full_content_rich_text_options,
+                template='wagtailpages/blocks/rich_text_block.html',
+            )),
+            ('card_grid', customblocks.CardGridBlock()),
+            ('image_grid', customblocks.ImageGridBlock()),
+            ('iframe', customblocks.iFrameBlock()),
+            ('image', customblocks.AnnotatedImageBlock()),
+            ('audio', customblocks.AudioBlock()),
+            ('image_text', customblocks.ImageTextBlock()),
+            ('image_text_mini', customblocks.ImageTextMini()),
+            ('video', customblocks.VideoBlock()),
+            ('linkbutton', customblocks.LinkButtonBlock()),
+            ('looping_video', customblocks.LoopingVideoBlock()),
+            ('pulse_listing', customblocks.PulseProjectList()),
+            ('single_quote', customblocks.SingleQuoteBlock()),
+            ('slider', customblocks.FoundationSliderBlock()),
+            ('spacer', customblocks.BootstrapSpacerBlock()),
+            ('airtable', customblocks.AirTableBlock()),
+            ('datawrapper', customblocks.DatawrapperBlock()),
+            ('typeform', customblocks.TypeformBlock()),
+        ),
+        block_counts={'typeform': {'max_num': 1}},
+        null=True,
+        blank=False,
+    )
+
+    content_panels = wagtail_models.Page.content_panels + [
+        image_panels.ImageChooserPanel('hero_image'),
+        panels.InlinePanel('author_profile_relations', heading='Authors', label='Author'),
+        panels.InlinePanel(
+            'content_category_relations',
+            heading='Content categories',
+            label='Content category',
+            max_num=2,
+        ),
+        panels.InlinePanel(
+            'related_article_relations',
+            heading='Related articles',
+            label='Article',
+            max_num=6,
+        ),
+        panels.StreamFieldPanel('body'),
+    ]
+
+    def get_context(self, request: http.HttpRequest, *args, **kwargs) -> dict:
+        context = super().get_context(request, *args, **kwargs)
+        context['home_page'] = self.get_parent().get_parent().specific
```

Comment:
** Suggestion (non-blocking)**

I'm a tad wary about this, as finding the homepage is depending on the position of self in the page tree. That might be fine given the page rules, another way to do it might be to find the `self.request > site > home_page` ?

Reply:
Yea, true. The `home_page` is actually not the site's homepage. The existing templates are using this unfortunately named variable to refer to the "buyersguide homepage" (the first page of that site structure). 

I have tried to make the retrieval a bit more robust by limiting the type of the page and hopefully made the code clearer by moving this into a method.

See 5a4b6b63a3af9274b5c579a41012acf03529ab8f

Reply:
Also a test 😅